website/static/js/registrationUtils.js

```diff
@@ -0,0 +1,802 @@
+            }).then(function() {
+                bootbox.dialog({
+                    message: messages.afterSubmitForApproval,
+                    title: 'Pre-Registration Prize Submission',
+                    buttons: {
+                        registrations: {
+                            label: 'Return to registrations page',
+                            className: 'btn-primary pull-right',
+                            callback: function() {
+                                window.location.href = self.draft().urls.registrations;
+                            }
+                        }
+                    }
+                });
+            }).fail($osf.growl.bind(null, 'Error submitting for review', language.submitForReviewFail));
+        }
+    });
+};
+/**
+ * Create a new draft
+ **/
+RegistrationEditor.prototype.create = function(schemaData) {
+    var self = this;
+
+    var metaSchema = self.draft().metaSchema;
+
+    return $osf.postJSON(self.urls.create, {
+        schema_name: metaSchema.name,
+        schema_version: metaSchema.version,
+        schema_data: schemaData
+    }).then(function(response) {
+        var draft = self.draft();
+        draft.pk = response.pk;
+        self.draft(draft);
+    });
+};
+
+/**
+ * Autosave the form state, immediately or deferred.
+ *
+ **/
+RegistrationEditor.prototype.autoSave = $osf.throttle(function () {
+    // TODO(lyndsysimon): Test this.
+    var self = this;
+    // Delay between the completion of a save and the next request
+    var delay = 5000;
+
+    if (!self.isDirty()) {
+        return;
+    }
+
+    var hasBeenSaved = function() {
+        // bool
+        return self.lastSaveRequest() !== undefined;
+    };
+    var delayRemaining = function () {
+        // milliseconds
+        return delay - self.lastSaveAge();
+    };
+    var delaySatisfied = function () {
+        // bool
+        return delayRemaining() <= 0;
+    };
+
+    if(!hasBeenSaved()) {
+        self.save().always(function() {
+            self.autoSave();
+        });
+    } else if (delaySatisfied()) {
+        self.save().always(function() {
+            self.autoSave();
+        });
+    } else {
+        window.setTimeout(self.autoSave.bind(self), delayRemaining());
+    }
+}, 1000);
+
+/**
+ * Save the current Draft
+ **/
+RegistrationEditor.prototype.save = function() {
+    // TODO(lyndsysimon): Test this.
+    var self = this;
+    var request;
+
+    var payload = self.serialized();
+    if (typeof self.draft().pk === 'undefined') {
+        // Draft has not yet been assigned a PK; create a new DraftRegistration
+        request = $osf.postJSON(self.urls.create, payload);
+        request.done(function(data) { self.draft().pk = data.pk; }.bind(self));
+    } else {
+        // Update an existing DraftRegistration
+        var url = self.urls.update.replace('{draft_pk}', self.draft().pk);
+        request = $osf.putJSON(url, payload);
+    }
+
+    self.lastSaveRequest(request);
```

Comment:
If I'm reading this correctly, it's possible to have two requests in flight at the same time if the user manually clicks the save button while autosave is running.

But that doesn't guarantee that the requests will be processed in the same order by the server, which could lead to data loss in edge cases.  I can think of a few options, but all of them feel like hacks on top of this scaffold; we might want to add this to the discussion tomorrow AM.